Automattic/jetpack
# WAF: Add a dedicated automatic-rules.php file for storing firewall rules

This PR ensures that sites do not remove their automatic rules previously downloaded from WPCOM, in situations where the rules file is being re-generated and the site no longer has access to the latest firewall rules (i.e. cancelled their Scan plan).

#### Changes proposed in this Pull Request:

* Adds a new `rules/automatic-rules.php` file for storing automatic rules fetched from WPCOM.
* Updates the generation of `rules/rules.php` to `require()` the relevant rules files, based on the WAF configuration (automatic and manual rules enabled options).
* Updates the Protect plugin to allow users to toggle automatic rules when they have the free plan but an old copy of `automatic-rules.php` available to use.

#### Other information:


#### Jetpack product discussion



peb6dq-p2-m9

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Start with a Jetpack site with a Scan plan.
* Activate the WAF module and ensure the `automatic-rules.php` file is generated, but not yet included in the main `rules.php` file.
* Validate a `?test=<script>` request is not blocked.
* Enable automatic rules.
* Validate a `?test=<script>` request is blocked.
* Enable manual rules and update the IP allow/block list with an IP address you can test.
* Validate the relevant IP rules file is updated, along with the main `rules.php` file.
* Validate the IP is blocked/allowed.
* Disable automatic rules.
* Validate that the relevant IP rules file is not updated, but no longer required by the main `rules.php` file.
* Validate the IP is no longer blocked/allowed.
* Cancel your Jetpack Scan plan.
* Toggle the WAF module off and then on again.
* Validate that the `automatic-rules.php` file still contains the automatic rules fetched at the beginning of your testing.

#### Screenshots

When the site previously upgraded and fetched automatic firewall rules, and then downgraded back to the free plan:

<img width="1042" alt="Screenshot 2023-01-04 at 11 30 39 AM" src="https://user-images.githubusercontent.com/10933065/210625120-8fd33110-e078-46f9-9fde-cb64218d9e2d.png">
